devmand/gateway/src/devmand/channels/cli/engine/Engine.cpp

```diff
@@ -114,6 +177,28 @@ shared_ptr<ModelRegistry> Engine::getModelRegistry() const {
   return mreg;
 }
 
+shared_ptr<DeviceContext> Engine::getDeviceContext(const DeviceType& type) {
+  return pluginRegistry->getDeviceContext(type);
+}
+
+// TODO should this be singleton/cached?
```

Comment:
Cached, yes per device type. Constructing ReaderRegistry / WriterRegistry is expensive

Reply:
Doesn't have to be done now

Reply:
Reworded TODO